libtsuba/include/tsuba/TxnContext.h

```diff
@@ -0,0 +1,101 @@
+#ifndef KATANA_LIBTSUBA_TSUBA_TXNCONTEXT_H_
+#define KATANA_LIBTSUBA_TSUBA_TXNCONTEXT_H_
+
+namespace tsuba {
+
+class KATANA_EXPORT TxnContext {
+public:
+  void InsertNodePropertyRead(std::string name) {
+    node_properties_read_.insert(name);
+  }
+
+  void InsertNodePropertyRead(std::unordered_set<std::string> names) {
+    node_properties_read_.insert(names.begin(), names.end());
+  }
+
+  void InsertNodePropertyRead(std::vector<std::string> names) {
+    node_properties_read_.insert(names.begin(), names.end());
+  }
+
+  void InsertNodePropertyWrite(std::string name) {
+    node_properties_write_.insert(name);
+  }
+
+  void InsertNodePropertyWrite(std::unordered_set<std::string> names) {
+    node_properties_write_.insert(names.begin(), names.end());
+  }
+
+  void InsertNodePropertyWrite(std::vector<std::string> names) {
+    node_properties_write_.insert(names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyRead(std::string name) {
+    edge_properties_read_.insert(name);
+  }
+
+  void InsertEdgePropertyRead(std::unordered_set<std::string> names) {
+    edge_properties_read_.insert(names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyRead(std::vector<std::string> names) {
+    edge_properties_read_.insert(names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyWrite(std::string name) {
+    edge_properties_write_.insert(name);
+  }
+
+  void InsertEdgePropertyWrite(std::unordered_set<std::string> names) {
+    edge_properties_write_.insert(names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyWrite(std::vector<std::string> names) {
+    edge_properties_write_.insert(names.begin(), names.end());
+  }
+
+  void SetAllPropertiesRead() { all_properties_read_ = true; }
+
+  void SetAllPropertiesWrite() { all_properties_write_ = true; }
+
+  void SetTopologyRead() { topology_read_ = true; }
+
+  void SetTopologyWrite() { topology_write_ = true; }
```

Comment:
Why do you need these setters?

Reply:
They are supposedly called when a function changes the topology of a graph (a global property). Also, the boolean can only be set from false to true but not reversely.